knees4bees/whats-cookin
# Feature/search by tag

--->  ((are you in the right branch??))
## What does this PR do?
- Creates filter-by-tag capability and combines it with search so that users can search among a subset of recipes (e.g., breakfast recipes that include chocolate).
- Adds tags to the list of things that are searched for using user's search bar input (previously just ingredient and recipe name).
- Removes separate search button (now just one "go" button).
- Adds a dynamic title to recipe list view to describe the list of recipes (e.g. "Breakfast recipes matching "chocolate" ").

## How should it be tested?
 - 1. Double-check that all tests are still passing (they were at the time of writing).
 - 2. In the UI, try selecting all categories from the dropdown, with and without something typed in the search bar, and confirm that the results make sense. Also try searching for a tag like "snack" or "birthday" that isn't included in the dropdown categories and make sure results appear.
